cli/azd/pkg/tools/docker/docker.go

```diff
@@ -23,6 +23,19 @@ type Docker struct {
 	commandRunner exec.CommandRunner
 }
 
+func (d *Docker) Login(ctx context.Context, loginServer string, username string, password string) error {
+	_, err := d.executeCommand(ctx, ".", "login",
+		"--username", username,
+		"--password", password,
```

Comment:
Could we use `--password-stdin` here instead? This way we don't leak the password to folks who might be looking at processes running on the system (for example, I know in Linux you can observe the command line arguments for other programs by looking in `/proc`).

Reply:
@wbreza pointed out that this strategy mirrors what the `az` CLI is doing today.  Let's stick with this for now but follow up with them to better understand if there was a specific reason they eschewed using `--password-stdin`.